daft/io/aws_config.py

```diff
@@ -0,0 +1,21 @@
+from typing import TYPE_CHECKING
+
+from daft.daft import S3Config
+
+if TYPE_CHECKING:
+    import boto3
+
+
+def boto3_client_from_s3_config(service: str, s3_config: S3Config) -> "boto3.client":
+    import boto3
```

Comment:
Do we have to amend our requirements to make boto3 a requirement when using deltalake then?